<launch>
    <!-- Start the gps acquisition and transformation -->
    <include file="$(find gps_transform)/launch/gps_transform.launch" />
    <node name="coord_gps" pkg="coord_gps" type="coord_gps" />

    <!-- Start the lidar -->
    <include file="$(find ydlidar_ros)/launch/X4.launch" />
    <node name="scan_detect_obstacle" pkg="scan_detect_obstacle" type="scan_detect_obstacle" />

    <!-- Start the arduino communication -->
    <arg name="arduino_port" default="ACM0" />
    <node name="arduino_comm" pkg="rosserial_python" type="serial_node.py" args="/dev/tty$(arg arduino_port)" />

    <!-- Missing vision process node -->
    <node name="detect_waste_node" pkg="detect_waste" type="detect_waste_node" />

    <!-- Start path finding -->
    <node name="path_finding_node" pkg="path_finding" type="path_finding_node" />
</launch>
